spec/async-spec-helpers.js

```diff
@@ -60,3 +62,9 @@ function waitsForPromise (fn) {
     })
   })
 }
+
+export function emitterEventPromise (emitter, event, timeout = 5000) {
+  let emitted = false
+  emitter.once(event, () => { emitted = true })
+  return until(`${event} is emitted`, () => emitted, timeout)
+}
```

Comment:
I think we could just resolve this promise explicitly when the event is emitted. The current code works but the polling seems unnecessary.

```js
export function emitterEventPromise (emitter, event, timeout = 5000) {
  return new Promise((resolve, reject) => {
    const timeoutHandle = setTimeout(reject, timeout)
    emitter.once(event, () => {
      clearTimeout(timeoutHandle)
      resolve()
    })
  })
}
```